src/app/new-page/component/content-flow-item/content-flow-item.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import {
+  Content,
+  FlowItem,
+  FlowWatcher
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+import { PageService } from '../../service/page-service.service';
+
+@Component({
+  selector: 'app-content-new-flow-item',
+  templateUrl: './content-flow-item.component.html',
+  styleUrls: ['./content-flow-item.component.css']
+})
+export class ContentFlowItemNewComponent implements OnChanges {
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input() item: FlowItem;
+  contents: Content[];
+  ready: boolean;
+  state: any;
+  isHidden: boolean;
+  flowWatcher: FlowWatcher;
+
+  constructor(private pageService: PageService) {
+    this.state = this.pageService.parserState();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.flowWatcher.close();
+  }
+
+  private init(): void {
+    this.isHidden = this.item.isGone(this.state);
```

Comment:
`watchIsGone` will emit the initial state when you start watching for changes. I don't know if the initial emission will trigger inline or as a subsequent event on the event loop though.

Reply:
so, setting `isHidden` here may not be necessary, but it doesn't actually hurt anything

Reply:
When `isGone` is called, it immediately updates `this.isHidden`. While `watchIsGone` has some delay before updating `this.isHidden` with the new value.

I understand `watchIsGone` is watching to see if there are changes to the item's `isGone` value, but I'm confused about how one can change the `isGone` value to trigger the `watchIsGone`.

Reply:
k, the delay is from it being called as as subsequent event on the event loop, so the initial setting of `isHidden` is appropriate.

How the state can be updated is actually a bit complex, simple answer is that visibility for flow items in the emoji survey change based on which items you select on the multiselect. In this case the multiselect & flow are on different screens, so the state doesn't update "live" because it's off screen. But if the multiselect and the flow were on the same screen, then toggling an option should live update the flow items.

I believe there is a test for this live updating somewhere in renderer testing.

Reply:
That makes sense. Thank you for explaining. I was able to test it on the renderer testing and can confirm it's working in realtime.

Reply:
I'll remove the function `isGone` and only use `watchIsGone`.